firestore/cloud-client/snippets.py

```diff
@@ -883,3 +883,65 @@ def update_document_increment(db):
 
     washington_ref.update("population", firestore.Increment(50))
     # [END fs_update_document_increment]
+
+
+# [START fs_counter_classes]
+import random
+from google.cloud import firestore
+
+
+class Shard(object):
+    """
+    Shard is a single counter, which is used in a group
+    of other shards within Counter.
+    """
+
+    def __init__(self):
+        self._count = 0
+
+    def to_dict(self):
+        return {"count": self._count}
+
+
+class Counter(object):
+    """
+    Counter is a collection of documents (shards)
```

Comment:
Related to other comment. 
`A counter stores a collection of shards which are summed to return a total count. This allows for more frequent incrementing than a single document.`